### ISSUE TITLE
Could the drag profile also include the lift coefficient (Cl*S)?

### ISSUE THREAD

**Issue by Arrowstar** [Ren0k/Project-Atmospheric-Drag#3]
I was going through the kOS code here this afternoon and noticed that it has most of what would be needed to also compute the lift coefficient, Cl\*S.  It would be great if the drag profile could also return this information.  In particular, if it returned Cl\*S for wings and Cl\*S for body lift parts separately, that would be awesome because the two lift vectors there are different in the vehicle reference frame.

The big impediment I notice now is that the function getVesselPartList() in Analysis.ks doesn't properly figure out the area to use for lift calculations.  This appears to be because it's looking for ModuleLiftingSurface on parts (Line 252), but this module isn't necessary for a part to produce lift.  It thus never marks parts as "body lift", and I guess there's no call to figure out what the appropriate area should be in lieu of that.

According to [this video](https://www.youtube.com/watch?v=_D2VmGk8bmM) the area to use is related to the drag cube face area times the pseudo drag coefficient for each part, but I don't know more than that.  See 9:50 into the video.

I'm looking to be able to use your wonderful work to pull out lift information as well as drag, so anything you could do would be super helpful in this regard!  Thank you!

**Comment by Ren0k** [comment]
### Why is lift used
For the purpose of this tool, lift is only relevant as lift can produce induced drag that adds to total drag.
That is why lift calculations are included; not to be used by the user, while in theory it can be exported.

### Exporting data
I do like your idea of making this exportable: I think this adds to your previous posts as well.
As much as I like it, this goes beyong the original idea of this tool.
The original goal was to provide the backbone to make atmospheric impact calculations possible.

### Lift and Body Lift
Have a look at the readme file, in particular the part about body lift.
For a body part to create lift it still needs to have a **ModuleLiftingSurface**, see all Mk2 spaceplane parts for example.
The body acts just like a wing, it just also has a dragcube that provides drag.

Now I think in the video provided, body lift refers to something else.
When a dragcube part has an angle to the relative airflow, a part of the drag (that normally works in the opposite direction of relative airflow) will have a 90 degree angle to the side as measured from the surface velocity vector.
Have a look at the readme file section **Lifting Surface Drag**, look at the image and read the section about **light blue arrows**.
Because sideways drag will not add or subtract from total drag, as it is always 90 degrees offset from the relative airflow, it's not used in the calculations in this tool.

### The Future
I like all ideas that you provide, but to keep this project close to the original goal, i think a new tool is required that could build on the foundations laid here. 
This is definitely something that I would like to do, but when the time is right.

**Comment by Arrowstar** [comment]
> Now I think in the video provided, body lift refers to something else.
When a dragcube part has an angle to the relative airflow, a part of the drag (that normally works in the opposite direction of relative airflow) will have a 90 degree angle to the side as measured from the surface velocity vector.
> Have a look at the readme file section Lifting Surface Drag, look at the image and read the section about light blue arrows.
> Because sideways drag will not add or subtract from total drag, as it is always 90 degrees offset from the relative airflow, it's not used in the calculations in this tool.

Okay, I think this is the part I didn't understand properly.  Do I understand correctly, then, that your code does not produce the full Cd*A values but instead only the portion that is directly opposing the atmosphere-relative velocity (meaning the portion which actually produces drag and not "lift")?

**Comment by Ren0k** [comment]
> > Now I think in the video provided, body lift refers to something else.
> > When a dragcube part has an angle to the relative airflow, a part of the drag (that normally works in the opposite direction of relative airflow) will have a 90 degree angle to the side as measured from the surface velocity vector.
> > Have a look at the readme file section Lifting Surface Drag, look at the image and read the section about light blue arrows.
> > Because sideways drag will not add or subtract from total drag, as it is always 90 degrees offset from the relative airflow, it's not used in the calculations in this tool.
> 
> Okay, I think this is the part I didn't understand properly. Do I understand correctly, then, that your code does not produce the full Cd*A values but instead only the portion that is directly opposing the atmosphere-relative velocity (meaning the portion which actually produces drag and not "lift")?

Yes, that is correct. Lift generated by lifting surfaces are important though, as they produce induced drag which acts in the direction of the reversed surface velocity vector. But lift acting perpendicular to the velocity vector is not relevant as this will not add or subtract from total drag. Same applies to drag cube force vectors.
